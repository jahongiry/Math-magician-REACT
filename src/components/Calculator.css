.container {
  background-color: rgb(224, 224, 224);
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  margin: auto;
  width: 30%;
  box-shadow:
    rgba(50, 50, 93, 0.25) 0 50px 100px -20px,
    rgba(0, 0, 0, 0.3) 0 30px 60px -30px,
    rgba(10, 37, 64, 0.35) 0 -2px 6px 0 inset;
}

.display {
  background-color: rgb(133, 133, 147);
  text-align: right;
  padding: 1.2rem 0.5rem;
  grid-column: 1/5;
}

.zero-button {
  grid-column: 1/3;
}

.orange-cl {
  border: rgb(133, 133, 147) 1px solid;
  background-color: rgb(245, 145, 62);
}

button {
  border: rgb(133, 133, 147) 1px solid;
  padding: 1.2rem 1.5rem;
}

.top-container {
  display: flex;
  padding: 5rem 2rem;
}

.side {
  font-size: 2rem;
}
