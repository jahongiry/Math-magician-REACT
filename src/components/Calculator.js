import { React, useState } from 'react';
import calculate from '../logic/calculate';
import './Calculator.css';

const Calculator = () => {
  const [state, setState] = useState({ total: 0, nextNum: 0 });

  const clickHandler = (e) => {
    const inputValue = e.target.value;
    const calculation = calculate(state, inputValue);
    setState(calculation);
  };

  const displayHandler = (e) => {
    setState({
      total: e.target.value,
    });
  };

  return (
    <div>
      <div className="container">
        <span className="display" onChange={displayHandler}>
          {state.next || state.total || 0}
        </span>
        <button type="submit" onClick={clickHandler} value="AC">
          AC
        </button>
        <button type="submit" onClick={clickHandler} value="+/-">
          +/-
        </button>
        <button type="submit" onClick={clickHandler} value="%">
          %
        </button>
        <button
          className="orange-cl"
          type="submit"
          onClick={clickHandler}
          value="รท"
        >
          รท
        </button>
        <button type="submit" onClick={clickHandler} value="7">
          7
        </button>
        <button type="submit" onClick={clickHandler} value="8">
          8
        </button>
        <button type="submit" onClick={clickHandler} value="9">
          9
        </button>
        <button
          className="orange-cl"
          type="submit"
          onClick={clickHandler}
          value="x"
        >
          x
        </button>
        <button type="submit" onClick={clickHandler} value="4">
          4
        </button>
        <button type="submit" onClick={clickHandler} value="5">
          5
        </button>
        <button type="submit" onClick={clickHandler} value="6">
          6
        </button>
        <button
          className="orange-cl"
          type="submit"
          onClick={clickHandler}
          value="-"
        >
          -
        </button>
        <button type="submit" onClick={clickHandler} value="1">
          1
        </button>
        <button type="submit" onClick={clickHandler} value="2">
          2
        </button>
        <button type="submit" onClick={clickHandler} value="3">
          3
        </button>
        <button
          className="orange-cl"
          type="submit"
          onClick={clickHandler}
          value="+"
        >
          +
        </button>
        <button
          className="zero-button"
          type="submit"
          onClick={clickHandler}
          value="0"
        >
          0
        </button>
        <button type="submit" onClick={clickHandler} value=".">
          .
        </button>
        <button
          className="orange-cl"
          type="submit"
          onClick={clickHandler}
          value="="
        >
          =
        </button>
      </div>
    </div>
  );
};

export default Calculator;
