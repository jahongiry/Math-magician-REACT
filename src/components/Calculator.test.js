import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

it('renders if there is Navigation component', () => {
  const three = renderer.create(<Calculator />).toJSON();
  expect(three).toMatchSnapshot();
});
