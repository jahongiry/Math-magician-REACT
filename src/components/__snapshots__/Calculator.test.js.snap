// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders if there is Navigation component 1`] = `
<div
  className="top-container"
>
  <div
    className="side"
  >
    <h2>
      Let's do some math!
    </h2>
  </div>
  <div
    className="container"
  >
    <span
      className="display"
      onChange={[Function]}
    >
      0
    </span>
    <button
      onClick={[Function]}
      type="submit"
      value="AC"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="+/-"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="%"
    >
      %
    </button>
    <button
      className="orange-cl"
      onClick={[Function]}
      type="submit"
      value="รท"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="7"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="8"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="9"
    >
      9
    </button>
    <button
      className="orange-cl"
      onClick={[Function]}
      type="submit"
      value="x"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="4"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="5"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="6"
    >
      6
    </button>
    <button
      className="orange-cl"
      onClick={[Function]}
      type="submit"
      value="-"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="1"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="2"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="3"
    >
      3
    </button>
    <button
      className="orange-cl"
      onClick={[Function]}
      type="submit"
      value="+"
    >
      +
    </button>
    <button
      className="zero-button"
      onClick={[Function]}
      type="submit"
      value="0"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="submit"
      value="."
    >
      .
    </button>
    <button
      className="orange-cl"
      onClick={[Function]}
      type="submit"
      value="="
    >
      =
    </button>
  </div>
</div>
`;
